### PR TITLE
Convert lib/devicetrust to use slog

### PR DESCRIPTION
**Pull request by rosstimothy**

### CODE REVIEW

File: lib/devicetrust/authz/authz.go [hunk 1]
```diff
@@ -45,8 +47,8 @@ func IsTLSDeviceVerified(ext *tlsca.DeviceExtensions) bool {
 
 // VerifyTLSUser verifies if the TLS identity has the required extensions to
 // fulfill the device trust configuration.
-func VerifyTLSUser(dt *types.DeviceTrust, identity tlsca.Identity) error {
-	return verifyDeviceExtensions(dt, identity.Username, IsTLSDeviceVerified(&identity.DeviceExtensions))
+func VerifyTLSUser(ctx context.Context, dt *types.DeviceTrust, identity tlsca.Identity) error {
```
Comment (codingllama) [inline lib/devicetrust/authz/authz.go:50]: I'm not sure passing the context here is the right call - it isn't respected or used in any way, apart from logging. I would log with context.Background() instead.

File: lib/devicetrust/authz/authz.go [hunk 2]
```diff
@@ -83,24 +85,22 @@ func HasDeviceTrustExtensions(extensions []string) bool {
 
 // VerifySSHUser verifies if the SSH certificate has the required extensions to
 // fulfill the device trust configuration.
-func VerifySSHUser(dt *types.DeviceTrust, cert *ssh.Certificate) error {
+func VerifySSHUser(ctx context.Context, dt *types.DeviceTrust, cert *ssh.Certificate) error {
```
Comment (codingllama) [inline lib/devicetrust/authz/authz.go:88]: Same for adding a context.

**Review by codingllama — COMMENTED** [review]
Thanks!

**Review by marcoandredinis — APPROVED** [review]

File: lib/devicetrust/authz/authz.go (hunk 1 above)
Comment (codingllama) [reply lib/devicetrust/authz/authz.go:50]: I'm not convinced that maybe augmenting a debug log is worth the change in signature. I would expect a method that takes a context to respect it, or at least use something that does. I don't think just logging qualifies.

Comment (codingllama) [reply lib/devicetrust/authz/authz.go:50]: While I don't disagree with the RFD (except for context.TODO() which I think is always a bad choice), I do disagree with this specific change. I would rather not log that particular line than change the signature in a way that gives a wrong impression about what the methods do.

**Review by codingllama — APPROVED** [review]

File: lib/devicetrust/authz/authz.go (hunk 1 above)
Comment (codingllama) [reply lib/devicetrust/authz/authz.go:50]: Approved - I won't block the refactor but as stated before I would very much prefer not to change the signatures in this file.

**Pull request merged**